src/sync/app.cpp

```diff
@@ -841,5 +842,53 @@ void App::link_user(std::shared_ptr<SyncUser> user,
     }
 
 
+RemoteMongoClient App::remote_mongo_client(const std::string& service_name)
+{
+    m_service_name = util::Optional<std::string>(service_name);
```

Comment:
This should set the service name on the copy of the App made below and the function should be `const`. Even if mutating the member here is harmless, it'll be confusing when trying to debug things.

Reply:
im no longer setting the service name here, I've also marked this function as const